misk/src/main/kotlin/misk/web/actions/NotFoundAction.kt

```diff
@@ -38,4 +50,40 @@ class NotFoundAction @Inject constructor() : WebAction {
       )
     }
   }
+
+  /** Generates a message for a client that hints what may have been wrong with their request. */
+  private fun notFoundMessage(path: String, httpCall: HttpCall) = buildString {
+    val boundActions = servletProvider.get().boundActions
+    val alternativeActions = boundActions.mapNotNull {
+      it.matchByUrl(httpCall.url)
+    }.filterNot { it.action.pathPattern.matchesWildcardPath }
+
+    append("Nothing found at /$path.\n\n")
+    append("Received:\n")
+    append("  ${httpCall.dispatchMechanism.method} /$path\n")
+    for (accept in httpCall.accepts()) {
+      append("  Accept: $accept\n")
+    }
+    val contentType = httpCall.contentType()
+    if (contentType != null) {
+      append("  Content-Type: ${httpCall.contentType()}\n")
+    }
+    if (alternativeActions.isNotEmpty()) {
+      append("\n")
+    }
+
+    for (i in 0 until alternativeActions.size) {
+      val alternative = alternativeActions[i]
+      val metadata = alternative.action.metadata
+      append("Alternative:\n")
+      append("  ${metadata.dispatchMechanism.method} ${metadata.pathPattern}\n")
+      for (requestContentType in metadata.requestMediaTypes) {
+        append("  Content-Type: $requestContentType\n")
+      }
+      append("  Accept: ${metadata.responseMediaType}\n")
+      if (i < alternativeActions.size - 1) {
```

Comment:
you could make your life easier by moving this newline to the beginning of the loop and removing the check on line 71

Reply:
Fixed!